plugins/filter_nest/nest.c

```diff
@@ -321,6 +359,24 @@ static inline bool is_kv_to_nest(msgpack_object_kv * kv,
         return false;
     }
 
+    mk_list_foreach_safe(head, tmp, &ctx->wildcard_excludes) {
+        wildcard_exclude = mk_list_entry(head, struct filter_nest_wildcard, _head);
+
+        if (wildcard_exclude->key_is_dynamic) {
+            /* This will negatively match "ABC123" with prefix "ABC*" */
+            if (strncmp(key, wildcard_exclude->key, wildcard_exclude->key_len) == 0) {
```

Comment:
`key` is not a null terminated string and this comparison can potentially fail. Since `klen` is set, I recommend comparing first `klen == wildcard_exclude->key_len` and if it passes then do the `strncmp()`